File: src/main/java/com/ndgndg91/ordermatchingengine/order/dto/response/OrderEntryResponse.java
package com.ndgndg91.ordermatchingengine.order.dto.response;

import com.ndgndg91.ordermatchingengine.order.OrderEntry;
import lombok.Getter;
import lombok.ToString;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@ToString
public final class OrderEntryResponse {
    private final String orderId;
    private final int shares;
    private final String priceType;
    private final BigDecimal price;
    private final LocalDateTime timestamp;
    private final boolean partialMatched;
    private final int currentShares;

    public OrderEntryResponse(final OrderEntry orderEntry) {
        this.orderId = orderEntry.getOrderId();
        this.shares = orderEntry.getShares();
        this.priceType = orderEntry.getPriceType().name();
        this.price = orderEntry.getPrice();
        this.timestamp = orderEntry.getTimestamp();
        this.partialMatched = orderEntry.isPartialMatched();
        this.currentShares = orderEntry.shares();
    }
}

File: OrderMatchingEngineKotlin/src/main/kotlin/com/ndgndg91/ordermatchingenginekotlin/order/dto/response/OrderEntryResponse.kt
package com.ndgndg91.ordermatchingenginekotlin.order.dto.response

import com.ndgndg91.ordermatchingenginekotlin.order.OrderEntry
import java.math.BigDecimal
import java.time.LocalDateTime

data class OrderEntryResponse(private val e: OrderEntry) {
    val orderId: String = e.orderId
    val shares: Int = e.shares
    val priceType: String = e.priceType.name
    val price: BigDecimal = e.price
    val timestamp: LocalDateTime = e.timestamp
    val partialMatched: Boolean = e.partialMatched
    val currentShares: Int = e.shares()
}

